### PR TITLE
Check working on master

### PR DESCRIPTION
**Pull request by yashuk803**
Create new script for check working on the master branch before commit. This is very convenient when you forget to create a new branch when cloning a repository.

### CODE REVIEW

File: src/pre-commit.check_working_on_master.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -e
+
+BRANCH_MASTER="master";
+
+if [[ $(git rev-parse --abbrev-ref HEAD) == $BRANCH_MASTER ]];
+then
+    echo "!!!WARNING!!! Now you working on the master branch."
```
Comment (greeflas) [inline src/pre-commit.check_working_on_master.sh]: Let's add some color :)

Change this line to `printf "\n$YELLOW!!!WARNING!!! Now you are working on the master branch.\n$CO"`, and don't forget to add constants with color-code at top of this script

```bash
YELLOW='\033[0;33m'
CO='\033[0m'
```

File: src/pre-commit.check_working_on_master.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -e
+
+BRANCH_MASTER="master";
```
Comment (greeflas) [inline src/pre-commit.check_working_on_master.sh]: `;` no needed after variable definition

```suggestion
BRANCH_MASTER="master"
```

File: src/pre-commit.check_working_on_master.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+if [[ $(git rev-parse --abbrev-ref HEAD) == $BRANCH_MASTER ]];
+then
+    echo "!!!WARNING!!! Now you working on the master branch."
+    exec < /dev/tty
+    while :
+    do
+        read -p "Are you really want to continue?  [y/n] " answer
+        case "$answer" in
+            y|yes)
+                echo -e  "Commit successful."
```
Comment (greeflas) [inline src/pre-commit.check_working_on_master.sh]: 1. `"Commit successful."` -> `"Commited successfully.\n"`
2. Please remove `-e` modifier, it's not needed here.

File: src/pre-commit.check_working_on_master.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                exit 0
+                ;;
+            n|no)
+                echo -e "Commit has failed."
```
Comment (greeflas) [inline src/pre-commit.check_working_on_master.sh]: 1. `Commit has failed."` -> `Commit was canceled."`
2. Please remove `-e` modifier, it's not needed here.

**Review by greeflas — CHANGES_REQUESTED** [review]
Please add this command to README.md. Thanks in advance!

File: src/pre-commit.check_working_on_master.sh (hunk 1 above)
Comment (yashuk803) [reply src/pre-commit.check_working_on_master.sh]: Ok. This is a good idea:) Added.

File: src/pre-commit.check_working_on_master.sh (hunk 2 above)
Comment (yashuk803) [reply src/pre-commit.check_working_on_master.sh]: removed

File: src/pre-commit.check_working_on_master.sh (hunk 3 above)
Comment (yashuk803) [reply src/pre-commit.check_working_on_master.sh]: 1. changed
2. removed

File: src/pre-commit.check_working_on_master.sh (hunk 4 above)
Comment (yashuk803) [reply src/pre-commit.check_working_on_master.sh]: 1. changed
2. removed

**Comment by greeflas** [comment]
@yashuk803 what about README.md?

**Comment by yashuk803** [comment]
I added description script  in README.md)

пт, 22 мар. 2019 г. в 18:50, Vladimir Kuprienko <notifications@github.com>:

**Comment by greeflas** [comment]
@yashuk803 Merged. Thanks for contributing!

**Comment by yashuk803** [comment]
You are welcom. Thank you too for allowing me to contribute to your project:)


сб, 23 мар. 2019 г. в 15:07, Vladimir Kuprienko <notifications@github.com>:

**Pull request merged**